Jobsway2goMvc/Controllers/JobsController.cs

```diff
@@ -222,6 +228,51 @@ public async Task<IActionResult> DeleteConfirmed(int id)
             return RedirectToAction(nameof(Index));
         }
 
+        public async Task<IActionResult> SaveToCollection(int? id)
+        {
+            if (id == null || _context.Jobs == null)
+            {
+                return NotFound();
+            }
+
+            var job = await _context.Jobs.FindAsync(id);
+            if (job == null)
+            {
+                return NotFound();
+            }
+
+            JobCollectionViewModel jobCollection = new JobCollectionViewModel();
+
+            jobCollection.Job = job;
+
+            var collections = _context.Collections
+                .Include(a => a.Jobs)
+                .Where(a => a.User.Id == HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value).ToList();
+            ViewBag.Collections = new SelectList(collections, "Id", "Name");
+            return View(jobCollection);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> SaveToCollection(Job job, Collection collection)
+        {
+            var collection1 = await _context.Collections.Include(a => a.Jobs).FirstOrDefaultAsync(a => a.Id == collection.Id);
+            var job1 = await _context.Jobs.FirstOrDefaultAsync(a => a.Id == job.Id);
+
+            if (collection1 != null && job1 != null)
+            {
```

Comment:
1. When creating int properties in model, rename parameters to `jobId` and `collectionId`
2. Rename lists from `collection1` to just `collection` and `job1` to just `job`
3. Check after each one if that object is null, return immediately and show message if so
4. Format the first line a little bit better, it is a long line. Separate to new lines at .Include and .FirstOrDefaultAsync